### DISCUSSION TITLE
401 Unauthorized while running new it block.

### DISCUSSION THREAD

**Discussion by MayurJ-QA** [cypress-io/cypress#15737] [Questions and Help]
I am using Cypress to test web application. Most pages behave completely identical to when I perform actions manually.

However, It works fine when I perform all actions in same it block using Cypress. But I have some pages that return a 401 - Unauthorized when I perform some actions through Cypress using different it block on same page and same super domain.
I have already switched between the Chrome and Electron browsers also tried by adding sufficient wait, tried disabling web security but had no luck so far.

I checked network requests, whenever there is new POST request from new it block, it throws 401

Does anyone have any idea what I could try to overcome this 401 problem? I can't find anything useful on google on the subject.

![image](https://user-images.githubusercontent.com/81743107/113265866-3db5d600-92f2-11eb-8bff-ebbb3d233b17.png)

**Comment by portyu9** [comment]
Hi @MayurJ-QA 

The screenshot confirms this is caused by Cypress test isolation.

Cypress clears cookies, `localStorage`, and `sessionStorage` between `it()` blocks. So if your authentication token/session is created in the first test, it is gone when the second test sends the POST request, which explains the `401`.

The fix is to restore authentication before each test, preferably with `cy.session()`:

```js
beforeEach(() => {
  cy.session('authenticated-user', () => {
    cy.login()
  })

  cy.visit('/your-page')
})
```

Or simply perform the login/setup in `beforeEach()`.

Do not make one `it()` depend on authentication established by another `it()`.

`wait()`, `chromeWebSecurity`, or changing browsers will not fix this—the second test needs its own restored authenticated session.

Please mark this answer as accepted if it helped.
